app/src/main/java/org/jellyfin/androidtv/ui/playback/VideoManager.java

```diff
@@ -464,7 +464,13 @@ public int getExoPlayerTrack(@Nullable MediaStreamType streamType, @Nullable Lis
                 if (trackType == chosenTrackType) {
                     if (groupInfo.isTrackSelected(i)) {
                         if (trackFormat.id != null) {
-                            int id = Integer.parseInt(trackFormat.id) - 1;
+                            int id;
+                            try {
+                                id = Integer.parseInt(trackFormat.id) - 1;
+                            } catch (NumberFormatException e) {
+                                Timber.d("failed to parse track ID [%s]", trackFormat.id);
+                                return -1;
+                            }
```

Comment:
I don't think we should ever assume the id is an integer at all. There must be a reason the ExoPlayer developers made it a string.

Reply:
It looks like mpeg-ts is the main culprit as far as using something other than an integer for the ID.
I've come across two variations so far:
* Menu ID/ID (this is what's seen in all of the crashes eg: "17103/221")
* codec:title (eg: aac:rus a1)

Here's a good example:
https://github.com/google/ExoPlayer/issues/8824#issuecomment-823184070

I don't know of any ways to match a MediaStream track to an Exoplayer track other than using ordering or matching a property like the ID. In this case, perhaps ordering could be used second option though it would open up the possibility of guessing wrong.

As it is, as long as the exception is caught, if the track matching fails it will fall-back to switching tracks by transcoding like it did before track selection was added (and the exo track selection code isn't used while transcoding).

Reply:
Would it be possible to return the index in the track group instead of the id?

Reply:
> Would it be possible to return the index in the track group instead of the id?

* From my testing, there is 1 track per `TrackGroup`, and 1 `TrackGroup` per `TrackGroupInfo`.
* The docs do say that `TrackGroup` should have its own ID
https://exoplayer.dev/doc/reference/com/google/android/exoplayer2/source/TrackGroup.html
  but trying to use it in the project says it doesn't exist. 

> Or maybe even use a string in our own code.

Some tracks do have labels, but not all do. And there's no guarantee that there wouldn't be multiple tracks with the label "english" for instance.